### PR TITLE
Add new official build pipeline

### PR DESCRIPTION
**Pull request by agocke**

### CODE REVIEW

File: eng/official.yml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  repositories:
+  - repository: 1ESPipelineTemplates
+    type: git
+    name: 1ESPipelineTemplates/1ESPipelineTemplates
+    ref: refs/tags/release
+extends:
+  template: v1/1ES.Official.PipelineTemplate.yml@1ESPipelineTemplates
+  parameters:
+    pool:
+      name: Azure-Pipelines-1ESPT-ExDShared
```
Comment (akoeplinger) [inline eng/official.yml]: this is the wrong pool, we should be using NetCore1ESPool-Svc-Internal here.

given this is only really used for the SDLSources stage you can also use this setting:
```
  parameters:
    sdl:
      sourceAnalysisPool:
        name: NetCore1ESPool-Svc-Internal
        image: 1es-windows-2022
        os: windows
```

File: eng/official.yml [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+          enablePublishBuildAssets: true
+          enableMicrobuild: true
+          jobs:
+          - job: Windows_NT
+            pool:
+              ${{ if eq(variables.officialBuild, 'false') }}:
+                vmImage: windows-latest
+              ${{ if eq(variables.officialBuild, 'true') }}:
+                name: NetCore1ESPool-Internal
+                demands: ImageOverride -equals 1es-windows-2019
```
Comment (akoeplinger) [inline eng/official.yml]: does this even work? it should have `image` and `os` keys

File: eng/official.yml [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            testResultsFormat: vstest
+          enablePublishBuildAssets: true
+          enableMicrobuild: true
+          jobs:
+          - job: Windows_NT
+            pool:
+              ${{ if eq(variables.officialBuild, 'false') }}:
+                vmImage: windows-latest
+              ${{ if eq(variables.officialBuild, 'true') }}:
+                name: NetCore1ESPool-Internal
```
Comment (akoeplinger) [inline eng/official.yml]: we should be using NetCore1ESPool-Svc-Internal for servicing branches

File: eng/official.yml [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+      jobs:
+      - template: /eng/common/templates/jobs/jobs.yml@self
+        parameters:
+          enableTelemetry: true
+          helixRepo: dotnet/linker
+          enablePublishUsingPipelines: true
+          enablePublishBuildArtifacts: true
+          ${{ if eq(variables.officialBuild, 'false') }}:
+            enablePublishTestResults: true
+            testResultsFormat: vstest
```
Comment (akoeplinger) [inline eng/official.yml]: we can probably clean up the pipeline to not run tests etc

**Review by akoeplinger — COMMENTED** [review]
is the cecil submodule update intentional?

**Comment by agocke** [comment]
@akoeplinger Trying to fix it. There's a package downgrade due to the submodule referencing net40

File: eng/official.yml (hunk 4 above)
Comment (agocke) [reply eng/official.yml]: True

**Review by sbomer — APPROVED** [review]
LGTM if it works

**Comment by agocke** [comment]
CG is complaining, so I think it works, but we will need to update some dependencies.

**Comment by agocke** [comment]
I'm going to merge this and come back for the dependency update in a second PR.

**Pull request merged**